SPECS/ldns/ldns-swig-4.2.patch

```diff
@@ -0,0 +1,60 @@
+SWIG_Python_str_AsChar removal in SWIG 4.2.0
+
+The replacement, SWIG_PyUnicode_AsUTF8AndSize, has different memory
+management requirements.
+
+Submitted upstream: <https://github.com/NLnetLabs/ldns/pull/232>
+
+diff -ur ldns-1.8.3.orig/ldns-1.8.3/contrib/python/ldns_rdf.i ldns-1.8.3/ldns-1.8.3/contrib/python/ldns_rdf.i
+--- ldns-1.8.3.orig/ldns-1.8.3/contrib/python/ldns_rdf.i	2022-08-15 12:38:22.000000000 +0200
++++ ldns-1.8.3/ldns-1.8.3/contrib/python/ldns_rdf.i	2024-01-26 11:28:52.983540469 +0100
```

Comment:
Can we do a few things:
- Apply the suggestion below to replace the `-p2` with `-p1` in the spec?
- Remove `%autopatch -p2` from the spec and add `-p1` into `%autosetup`?
- Remove `-N` from `%autosetup`?

Most patches come in a form, which makes them applicable with `-p1`, so it's probably going to make maintenance easier and keep the spec more consistent with most of our other ones.
```suggestion
--- ldns-1.8.3/contrib/python/ldns_rdf.i	2022-08-15 12:38:22.000000000 +0200
+++ ldns-1.8.3/contrib/python/ldns_rdf.i	2024-01-26 11:28:52.983540469 +0100
```